add tests and backend structure

200kgsquat opened the pull request:
-routers (health, classify, test)
-tests for models and tokenizers
-updated classes add methods

200kgsquat committed:
add tests and backend structure

-routers (health, classify, test)
-tests for models and tokenizers
-updated classes add methods

ernestknurov commented on src/app/core/classifiers/tf_idf.py:
```diff
@@ -1,8 +1,16 @@
 from sklearn.metrics import accuracy_score
+from src.app.core.tokenizers.tokenizer_tf_idf import TextCleaner
+
+text_cleaner = TextCleaner()
+
+def tokenizer_func(text):
+    return text_cleaner.clean_text(text)
```
I don't see that you use this function somewhere in the code, so what is the purpose of this function?

ernestknurov commented on src/app/core/ner/bert_ner.py:
```diff
@@ -52,190 +46,6 @@ def __init__(self, model_name="dslim/bert-base-NER", label_list=None, device=Non
 
         self.ner_pipeline = None
 
```
Why did you remove train functionality?

ernestknurov commented on src/app/config/config.py:
```diff
@@ -1,18 +1,14 @@
-# Model files
 TFIDF_MODEL_PATH = os.path.join(MODELS_DIR, 'tfidf_logreg_pipeline.joblib')
-BERT_MODEL_PATH = os.path.join(MODELS_DIR, 'bert_model.pth')
+BERT_MODEL_PATH = os.path.join(MODELS_DIR, 'bert_model')  # папка с токенайзером и весами
```
please remove russian comments

ernestknurov commented on src/app/main.py:
```diff
@@ -1,29 +1,8 @@
-    return SkillsResponse(skills=['Python', 'Machine Learning'])
+from fastapi import FastAPI
+from src.app.api.routers import classify, extract_skills, health
 
```
where is lifespan?

ernestknurov commented on src/app/api/routers/health.py:
```diff
@@ -0,0 +1,12 @@
+# src/app/api/routers/health.py
```
remove this line, vibecoder

ernestknurov commented on src/app/api/routers/extract_skills.py:
```diff
@@ -0,0 +1,16 @@
+class NERRequest(BaseModel):
+    texts: list[str]
+
+class NERResponse(BaseModel):
```
schemas should be in schemas folder.

ernestknurov replied:
What is the purpose of `SkillsResponse` in schemas? I See you don't use it

ernestknurov commented on src/app/api/routers/classify.py:
```diff
@@ -0,0 +1,16 @@
+class ClassifyRequest(BaseModel):
+    texts: list[str]
+
+class ClassifyResponse(BaseModel):
```
Move it to schemas.

ernestknurov commented on src/app/api/dependencies/client.py:
```diff
@@ -0,0 +1,48 @@
+# src/app/api/dependencies/client.py
```
remove line 1

ernestknurov commented on src/app/api/dependencies/client.py:
```diff
@@ -0,0 +1,48 @@
+    return bert_model
+
+def get_ner_model():
+    return ner_model
```
Models should be initialised in the lifespan.

ernestknurov replied:
Why do you have tf-idf and bert? Choose one

ernestknurov commented on test/api/routers/test_classify.py:
```diff
@@ -0,0 +1,39 @@
+# tests/api/test_classify.py
```
remove cringe line in the beginning

ernestknurov reviewed (changes requested):
I highly doubt that you tested your application and it worked ok. Please address all the issues I mentioned. Also please make more detailed pr description. Also you forgot to add responses. Check notion guide.

200kgsquat committed:
adjustments from last rewiev

-deleted cringe lines
-add config paths to main.py
-retrain tf_idf because there was some troubles with saving
-add responses and got rid of validation in routers
-add lifespan that load models when we run application

ernestknurov commented on src/app/api/responses/classify.py:
```diff
@@ -0,0 +1,5 @@
+from pydantic import BaseModel
+from typing import List
+
+class ClassifyResponse(BaseModel):
+    predictions: List[str]
```
Responses should contain a dict of status codes and examples of the responses.

<img width="707" height="675" alt="Image" src="https://github.com/user-attachments/assets/aa9d11c9-a935-41fd-ba04-86258495f842" />

ernestknurov commented on src/app/api/routers/classify.py:
```diff
@@ -0,0 +1,15 @@
+
+router = APIRouter()
+
+@router.post("/classify", response_model=ClassifyResponse)
```
response_model => responses

ernestknurov commented on src/app/config/config.py:
```diff
@@ -1,18 +1,19 @@
 
-# Base paths
-BASE_DIR = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+BASE_DIR = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', '..', '..'))
```
make it simpler: base_dir = "../models"

ernestknurov commented on src/app/main.py:
```diff
@@ -1,29 +1,60 @@
+    global ner_model, tfidf_model
+
+    # --- Startup ---
+    print("Loading models...")
```
replace with logger

ernestknurov commented on src/app/main.py:
```diff
@@ -1,29 +1,60 @@
+    print("Loading models...")
+
+    # Load NER model
+    ner_model = BertNER()
+    try:
+        print(f"NER model initialized from {NER_MODEL_PATH}")
+    except Exception:
+        print("NER model not found, using default untrained model")
```
Move ner_model initialisation inside try/except block

ernestknurov commented on src/app/main.py:
```diff
@@ -1,29 +1,60 @@
+from src.app.core.tokenizers.tokenizer_tf_idf import tokenizer_func
+
+# Global variables to hold models
+ner_model: BertNER | None = None
+tfidf_model: TFIDFWrapper | None = None
```
why do you need this? Remove it

ernestknurov commented on src/app/main.py:
```diff
@@ -1,29 +1,60 @@
+    FastAPI lifespan for initializing models on startup
+    and cleaning up on shutdown.
+    """
+    global ner_model, tfidf_model
```
remove

ernestknurov commented on src/app/main.py:
```diff
@@ -1,29 +1,60 @@
+    ner_model = None
+    tfidf_model = None
+    app.state.ner_model = None
+    app.state.tfidf_model = None
```
useless code

200kgsquat committed:
adjustments from rewiev

-worked on responses
-remove useless code from main
